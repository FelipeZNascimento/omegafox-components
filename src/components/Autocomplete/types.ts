export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
