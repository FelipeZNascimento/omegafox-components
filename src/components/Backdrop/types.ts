export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
