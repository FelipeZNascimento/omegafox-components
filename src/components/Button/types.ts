export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
