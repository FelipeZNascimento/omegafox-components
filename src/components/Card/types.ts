export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
