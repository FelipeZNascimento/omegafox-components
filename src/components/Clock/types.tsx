import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
