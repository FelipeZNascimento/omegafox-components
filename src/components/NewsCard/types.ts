export interface INewsCardProps {
  title: string;
  resume: string;
  link: string;
  image: string;
  date: string;
}
