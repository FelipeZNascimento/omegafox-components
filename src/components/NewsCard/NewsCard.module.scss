@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include marged(all, s);
  @include shadowed($color-black, 0.4);

  border-radius: 16px;
  cursor: pointer;
  height: 240px;
  position: relative;
  transition: transform $transition-time;
  width: 160px;

  &:hover {
    transform: scale(1.2);
    transition: transform $transition-time;
    z-index: 99;
  }

  &:hover .resume {
    display: -webkit-box;
    opacity: 1;
  }

  &:hover .titleContainer {
    height: 80%;
  }

  &:hover .date {
    opacity: 1;
  }

  img {
    border-radius: 16px;
    height: 100%;
    left: 0;
    object-fit: cover;
    overflow: hidden;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }

  .titleContainer {
    @include bg-rgba-colored($color-grey5, 0.8);
    @include padded(all, s2);
    @include flex-aligned(center, center);

    border-bottom-left-radius: 16px;
    border-bottom-right-radius: 16px;
    bottom: 0;
    flex-direction: column;
    height: 40%;
    left: 0;
    position: absolute;
    right: 0;
    transition: height $transition-time;
  }

  .title {
    @include flex-aligned(center, center);
    @include font-colored($color-grey1);
    @include font-sized(m);
    @include marged(all, none);

    display: -webkit-box;
    font-weight: bold;
    overflow: hidden;
    -webkit-line-clamp: 5;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
  }

  .resume {
    @include flex-aligned(center, center);
    @include font-sized(s);
    @include font-colored($color-grey2);
    @include marged(all, none);
    @include padded(top, s);

    display: none;
    opacity: 0;
    overflow: hidden;
    -webkit-line-clamp: 5;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
  }

  .date {
    @include aligned(center);
    @include font-sized(s);
    @include font-colored($color-grey2);

    opacity: 0;
    position: absolute;
    bottom: 0;
    width: 100%;
  }
}
