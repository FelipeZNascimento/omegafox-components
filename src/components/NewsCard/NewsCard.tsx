import React from 'react';

import { INewsCardProps } from './types';
import styles from './NewsCard.module.scss';

export const NewsCard = ({
  title,
  resume,
  date,
  link,
  image
}: INewsCardProps) => {
  if (!image) {
    return null;
  }

  return (
    <a href={link} target="_blank" rel="noreferrer">
      <div className={styles.container}>
        <img src={image} alt={resume} />
        <div className={styles.titleContainer}>
          <p className={styles.title}>{title}</p>
          <p className={styles.resume}>{resume}</p>
          <div className={styles.date}>{date}</div>
        </div>
      </div>
    </a>
  );
};
