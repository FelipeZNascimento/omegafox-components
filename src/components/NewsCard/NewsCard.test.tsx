import React from 'react';
import { render } from '@testing-library/react';
import { NewsCard } from './NewsCard';
import { INewsCardProps } from './types';

describe('<NewsCard />', () => {
  const renderComponent = ({
    title,
    resume,
    date,
    link,
    image
  }: INewsCardProps) =>
    render(
      <NewsCard
        title={title}
        resume={resume}
        date={date}
        link={link}
        image={image}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      title: 'title',
      resume: 'resume',
      date: 'date ago',
      link: '#',
      image:
        'https://s2.glbimg.com/8_XbuLD2d1XCpSP9pUOw4sc7CdY=/540x304/top/smart/filters:max_age(3600)/https://s01.video.glbimg.com/deo/vi/27/10/10971027'
    });

    expect(container).toMatchSnapshot();
  });
});
