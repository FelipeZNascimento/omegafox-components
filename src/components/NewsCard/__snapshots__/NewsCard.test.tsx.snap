// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsCard /> should render 1`] = `
<div>
  <a
    href="#"
    rel="noreferrer"
    target="_blank"
  >
    <div
      class="container"
    >
      <img
        alt="resume"
        src="https://s2.glbimg.com/8_XbuLD2d1XCpSP9pUOw4sc7CdY=/540x304/top/smart/filters:max_age(3600)/https://s01.video.glbimg.com/deo/vi/27/10/10971027"
      />
      <div
        class="titleContainer"
      >
        <p
          class="title"
        >
          title
        </p>
        <p
          class="resume"
        >
          resume
        </p>
        <div
          class="date"
        >
          date ago
        </div>
      </div>
    </div>
  </a>
</div>
`;
